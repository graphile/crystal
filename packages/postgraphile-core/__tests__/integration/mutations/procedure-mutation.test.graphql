mutation {
  jsonIdentityMutation(input: { json: "{\"a\":1,\"b\":2,\"c\":3}" }) { json }
  jsonbIdentityMutation(input: { json: "{\"a\":1,\"b\":2,\"c\":3}" }) { json }
  jsonIdentityMutationIssue85: jsonIdentityMutation(input: { json: "[{\"amount\":\"44\"},{\"amount\":null}]" }) { json }
  jsonbIdentityMutationIssue85: jsonbIdentityMutation(input: { json: "[{\"amount\":\"44\"},{\"amount\":null}]" }) { json }
  jsonbIdentityMutationPlpgsqlIssue85: jsonbIdentityMutationPlpgsql(input: { _theJson: "[{\"amount\":\"44\"},{\"amount\":null}]" }) { json }
  jsonbIdentityMutationPlpgsqlWithDefaultIssue85: jsonbIdentityMutationPlpgsqlWithDefault(input: {}) { json }
  jsonbIdentityMutationPlpgsqlOverridingDefaultIssue85: jsonbIdentityMutationPlpgsqlWithDefault(input: { _theJson: "[{\"amount\":\"44\"},{\"amount\":null}]" }) { json }
  add1Mutation(input: { arg0: 1, arg1: 2 }) { clientMutationId integer }
  add2Mutation(input: { clientMutationId: "hello", a: 2, b: 2 }) { clientMutationId integer }
  add3Mutation(input: { clientMutationId: "world", arg1: 5 }) { clientMutationId integer }
  add4Mutation(input: { arg0: 1, b: 3 }) { integer }
  add4MutationError(input: { arg0: 1, b: 3 }) { integer }
  mult1(input: { arg0: 0, arg1: 1 }) { integer }
  mult2(input: { arg0: 1, arg1: 1 }) { integer }
  mult3(input: { arg0: 1, arg1: 2 }) { integer }
  mult4(input: { arg0: 5, arg1: 2 }) { integer }
  typesMutation(input: { a: "50", b: false, c: "xyz", d: [1, 2, 3], e: "{\"a\":1,\"b\":2,\"c\":3}", f: { start: { value: 1, inclusive: false }, end: { value: 5, inclusive: false } } }) { boolean }
  compoundTypeMutation(input: { object: { a: 419, b: "easy cheesy baked potatoes", c: RED, e: FOO_BAR, f: _EMPTY_, g: {hours: 5 }, fooBar: 8 } }) { compoundType { a b c d e f g { seconds minutes hours } fooBar } }
  compoundTypeSetMutation(input: { object: { a: 419, b: "easy cheesy baked potatoes", c: RED, e: FOO_BAR, f: _EMPTY_, g: {hours: 5}, fooBar: 8 } }) { compoundTypes { a b c d e f g { seconds minutes hours } fooBar } }
  compoundTypeArrayMutation(input: { object: { a: 419, b: "easy cheesy baked potatoes", c: RED, e: FOO_BAR, f: _EMPTY_, g: { hours: 5}, fooBar: 8 } }) { compoundTypes { a b c d e f g { seconds minutes hours } fooBar } }
  tableMutation(input: { id: 5 }) { post { nodeId id headline authorId } personByAuthorId { id name } postEdge { cursor node { id headline } } }
  tableMutationReturningNull: tableMutation(input: { id: -1 }) { post { nodeId id headline authorId } personByAuthorId { id name } postEdge { cursor node { id headline } } }
  tableSetMutation(input: {}) { people { name } }
  intSetMutation(input: { x: 5, z: 6 }) { integers }
  noArgsMutation(input: { clientMutationId: "x" }) { clientMutationId integer }
  returnVoidMutation(input: {}) { __typename }
  guidFn(input: {g: null}) { guid }
  guidFn2: guidFn(input: {g: "0123456789abcde"}) { guid }
  postMany(input: {posts: [
    {id: 7, headline: "headline", body: "body", authorId: 9000, enums: AWAITING, comptypes: {schedule: "2009-10-24 10:23:54+02", isOptimised: true}},
    {id: 8, headline: "headline", body: "body", authorId: 9000, enums: AWAITING, comptypes: {schedule: "2009-10-24 10:23:54+02", isOptimised: null}},
    {id: 9, headline: "headline", body: "body", authorId: 9000, enums: AWAITING, comptypes: {schedule: "2009-10-24 10:23:54+02", isOptimised: false}}
  ]}) { posts { id headline comptypes { schedule isOptimised } } }
  postWithSuffix(input: {
    post: {id: 15, headline: "headline_", body: "body"},
    suffix: "test"
  }) { post { id headline } }
  issue756Mutation(input:{}) { issue756 { id ts } }
  issue756SetMutation(input:{}) { issue756S { id ts } }

  mutationCompoundTypeArray(input: {object: {a: 419, b: "easy cheesy baked potatoes", c: RED, e: BAR_FOO, f: _EMPTY_, g: {hours: 5}, fooBar: 8}}) { compoundTypes { a b c d e f g { seconds minutes hours } fooBar } }
  mutationTextArray(input: {}) { strings }
  mutationIntervalArray(input: {}) { intervals { seconds minutes hours days months years } }
  mutationIntervalSet(input: {}) { intervals { seconds minutes hours days months years } }
}
