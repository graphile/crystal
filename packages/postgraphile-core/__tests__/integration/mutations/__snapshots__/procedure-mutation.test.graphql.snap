// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute 1`] = `
Object {
  "data": Object {
    "add1Mutation": Object {
      "clientMutationId": null,
      "integer": 3,
    },
    "add2Mutation": Object {
      "clientMutationId": "hello",
      "integer": 4,
    },
    "add3Mutation": Object {
      "clientMutationId": "world",
      "integer": null,
    },
    "add4Mutation": Object {
      "integer": 4,
    },
    "add4MutationError": null,
    "compoundTypeArrayMutation": Object {
      "compoundTypes": Array [
        Object {
          "a": 419,
          "b": "easy cheesy baked potatoes",
          "c": "RED",
          "d": null,
          "e": "FOO_BAR",
          "f": "_EMPTY_",
          "fooBar": 8,
          "g": Object {
            "hours": 5,
            "minutes": null,
            "seconds": null,
          },
        },
        null,
        Object {
          "a": 420,
          "b": "easy cheesy baked potatoes",
          "c": "RED",
          "d": null,
          "e": "FOO_BAR",
          "f": "_EMPTY_",
          "fooBar": 8,
          "g": Object {
            "hours": 5,
            "minutes": null,
            "seconds": null,
          },
        },
      ],
    },
    "compoundTypeMutation": Object {
      "compoundType": Object {
        "a": 420,
        "b": "easy cheesy baked potatoes",
        "c": "RED",
        "d": null,
        "e": "FOO_BAR",
        "f": "_EMPTY_",
        "fooBar": 8,
        "g": Object {
          "hours": 5,
          "minutes": null,
          "seconds": null,
        },
      },
    },
    "compoundTypeSetMutation": Object {
      "compoundTypes": Array [
        Object {
          "a": 419,
          "b": "easy cheesy baked potatoes",
          "c": "RED",
          "d": null,
          "e": "FOO_BAR",
          "f": "_EMPTY_",
          "fooBar": 8,
          "g": Object {
            "hours": 5,
            "minutes": null,
            "seconds": null,
          },
        },
        Object {
          "a": 420,
          "b": "easy cheesy baked potatoes",
          "c": "RED",
          "d": null,
          "e": "FOO_BAR",
          "f": "_EMPTY_",
          "fooBar": 8,
          "g": Object {
            "hours": 5,
            "minutes": null,
            "seconds": null,
          },
        },
      ],
    },
    "guidFn": Object {
      "guid": null,
    },
    "guidFn2": Object {
      "guid": "0123456789abcde",
    },
    "intSetMutation": Object {
      "integers": Array [
        1,
        2,
        3,
        4,
        5,
        null,
        6,
      ],
    },
    "issue756Mutation": Object {
      "issue756": null,
    },
    "issue756SetMutation": Object {
      "issue756S": Array [
        Object {
          "id": 1,
          "ts": "1999-06-10T20:00:00-04:00",
        },
        null,
        Object {
          "id": 2,
          "ts": "1999-06-10T20:00:00-04:00",
        },
      ],
    },
    "jsonIdentityMutation": Object {
      "json": "{\\"a\\":1,\\"b\\":2,\\"c\\":3}",
    },
    "jsonIdentityMutationIssue85": Object {
      "json": "[{\\"amount\\":\\"44\\"},{\\"amount\\":null}]",
    },
    "jsonbIdentityMutation": Object {
      "json": "{\\"a\\":1,\\"b\\":2,\\"c\\":3}",
    },
    "jsonbIdentityMutationIssue85": Object {
      "json": "[{\\"amount\\":\\"44\\"},{\\"amount\\":null}]",
    },
    "jsonbIdentityMutationPlpgsqlIssue85": Object {
      "json": "[{\\"amount\\":\\"44\\"},{\\"amount\\":null}]",
    },
    "jsonbIdentityMutationPlpgsqlOverridingDefaultIssue85": Object {
      "json": "[{\\"amount\\":\\"44\\"},{\\"amount\\":null}]",
    },
    "jsonbIdentityMutationPlpgsqlWithDefaultIssue85": Object {
      "json": "[]",
    },
    "mult1": Object {
      "integer": 0,
    },
    "mult2": Object {
      "integer": 1,
    },
    "mult3": Object {
      "integer": 2,
    },
    "mult4": Object {
      "integer": 10,
    },
    "mutationCompoundTypeArray": Object {
      "compoundTypes": Array [
        Object {
          "a": 419,
          "b": "easy cheesy baked potatoes",
          "c": "RED",
          "d": null,
          "e": "BAR_FOO",
          "f": "_EMPTY_",
          "fooBar": 8,
          "g": Object {
            "hours": 5,
            "minutes": null,
            "seconds": null,
          },
        },
        null,
        Object {
          "a": 420,
          "b": "easy cheesy baked potatoes",
          "c": "RED",
          "d": null,
          "e": "BAR_FOO",
          "f": "_EMPTY_",
          "fooBar": 8,
          "g": Object {
            "hours": 5,
            "minutes": null,
            "seconds": null,
          },
        },
      ],
    },
    "mutationIntervalArray": Object {
      "intervals": Array [
        Object {
          "days": null,
          "hours": null,
          "minutes": null,
          "months": null,
          "seconds": 12,
          "years": null,
        },
        Object {
          "days": null,
          "hours": 3,
          "minutes": null,
          "months": null,
          "seconds": null,
          "years": null,
        },
        Object {
          "days": null,
          "hours": 9,
          "minutes": 36,
          "months": null,
          "seconds": 7,
          "years": null,
        },
      ],
    },
    "mutationIntervalSet": Object {
      "intervals": Array [
        Object {
          "days": null,
          "hours": null,
          "minutes": null,
          "months": null,
          "seconds": 12,
          "years": null,
        },
        Object {
          "days": null,
          "hours": 3,
          "minutes": null,
          "months": null,
          "seconds": null,
          "years": null,
        },
        Object {
          "days": null,
          "hours": 9,
          "minutes": 36,
          "months": null,
          "seconds": 7,
          "years": null,
        },
      ],
    },
    "mutationTextArray": Object {
      "strings": Array [
        "str1",
        "str2",
        "str3",
      ],
    },
    "noArgsMutation": Object {
      "clientMutationId": "x",
      "integer": 2,
    },
    "postMany": Object {
      "posts": Array [
        Object {
          "comptypes": Array [
            Object {
              "isOptimised": true,
              "schedule": "2009-10-24T04:23:54-04:00",
            },
          ],
          "headline": "headline",
          "id": 7,
        },
        Object {
          "comptypes": Array [
            Object {
              "isOptimised": null,
              "schedule": "2009-10-24T04:23:54-04:00",
            },
          ],
          "headline": "headline",
          "id": 8,
        },
        Object {
          "comptypes": Array [
            Object {
              "isOptimised": false,
              "schedule": "2009-10-24T04:23:54-04:00",
            },
          ],
          "headline": "headline",
          "id": 9,
        },
      ],
    },
    "postWithSuffix": Object {
      "post": Object {
        "headline": "headline_test",
        "id": 15,
      },
    },
    "returnVoidMutation": Object {
      "__typename": "ReturnVoidMutationPayload",
    },
    "tableMutation": Object {
      "personByAuthorId": Object {
        "id": 5,
        "name": "Joe Tucker",
      },
      "post": Object {
        "authorId": 5,
        "headline": "Please, Don-Bot… look into your hard drive, and open your mercy file!",
        "id": 5,
        "nodeId": "WyJwb3N0cyIsNV0=",
      },
      "postEdge": Object {
        "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFs1XV0=",
        "node": Object {
          "headline": "Please, Don-Bot… look into your hard drive, and open your mercy file!",
          "id": 5,
        },
      },
    },
    "tableMutationReturningNull": Object {
      "personByAuthorId": null,
      "post": null,
      "postEdge": null,
    },
    "tableSetMutation": Object {
      "people": Array [
        Object {
          "name": "John Smith",
        },
        Object {
          "name": "Sara Smith",
        },
        Object {
          "name": "Budd Deey",
        },
        Object {
          "name": "Kathryn Ramirez",
        },
        Object {
          "name": "Joe Tucker",
        },
        Object {
          "name": "Twenty Seventwo",
        },
      ],
    },
    "typesMutation": Object {
      "boolean": false,
    },
  },
  "errors": Array [
    [GraphQLError: Deliberate error],
  ],
}
`;
